roles/preflight-checks/tasks/ceph.yml

```diff
@@ -0,0 +1,26 @@
+---
+# this yml is executed with ceph installed
+- name: check ceph health
+  command: ceph -s
+  register: result_ceph_status
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+
+- name: ceph status is not health ok
+  fail: msg="ceph status is not health ok"
+  when: not result_ceph_status.stdout |search("HEALTH_OK")
+
+- name: fail if no right cinder type
+  shell: . /root/stackrc; cinder type-list |grep CEPH
```

Comment:
You grep for CEPH. If it's found, you'll get a return code of 0, which means success. But it seems you want this to be a failure state, so you'll need to adjust the grep return code, or use a `failed_when` conditional.